### DISCUSSION TITLE
Reading ICC tag values

### DISCUSSION THREAD

**Discussion by sowa705** [SixLabors/ImageSharp#2852] [Q&A]
Hi, I'm trying to read ICC profile tag data entries and I've noticed that types like IccXyzTagDataEntry are `internal sealed` for some reason. What is the proper way to actually get data out of an ICC profile?

**Comment by JimBobSquarePants** [comment]
It's all internal currently while we work out the mechanics to allow translation between ICC profiles. See #1567

Once we have that nailed, we may expose them once we're confident in the API. What is your use case?

**Reply by sowa705** [reply]
Hi, my use case was loading matrices for color conversion between various color spaces (my application requires all images in display P3), I've currently bypassed this issue by using reflection to pull out the data from sealed classes but it's not optimal :)
